src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs

```diff
@@ -156,86 +155,24 @@ public static ValidationType GetValidation(CSharpType type, RequestLocation requ
-            static bool IsApiVersionParameter(RequestParameter requestParameter)
-                => requestParameter.Origin == "modelerfour:synthesized/api-version";
-            static bool IsEndpointParameter(RequestParameter requestParameter)
-                => requestParameter.Origin == "modelerfour:synthesized/host";
-            static bool IsContentTypeParameter(RequestParameter requestParameter)
-                => requestParameter.Origin == "modelerfour:synthesized/content-type";
-        }
-
-        private static RequestLocation GetRequestLocation(RequestParameter requestParameter)
-            => requestParameter.In switch
-            {
-                HttpParameterIn.Uri => RequestLocation.Uri,
-                HttpParameterIn.Path => RequestLocation.Path,
-                HttpParameterIn.Query => RequestLocation.Query,
-                HttpParameterIn.Header => RequestLocation.Header,
-                HttpParameterIn.Body => RequestLocation.Body,
-                _ => RequestLocation.None
-            };
-
-        private static FormattableString CreateDescription(RequestParameter requestParameter, CSharpType type, Constant? defaultValue = null)
-        {
-            FormattableString description = string.IsNullOrWhiteSpace(requestParameter.Language.Default.Description) ?
+            FormattableString description = string.IsNullOrWhiteSpace(requestParameter.Description) ?
                 (FormattableString)$"The {type:C} to use." :
-                $"{BuilderHelpers.EscapeXmlDocDescription(requestParameter.Language.Default.Description)}";
+                $"{BuilderHelpers.EscapeXmlDocDescription(requestParameter.Description)}";
             if (defaultValue != null)
             {
                 var defaultValueString = defaultValue?.Value is string s ? $"\"{s}\"" : $"{defaultValue?.Value}";
                 description = $"{description}{(description.ToString().EndsWith(".") ? "" : ".")} The default value is {defaultValueString}";
             }
 
-            return requestParameter.Schema switch
+            return requestParameter.Type switch
             {
-                ChoiceSchema choiceSchema when type.IsFrameworkType => AddAllowedValues(description, choiceSchema.Choices),
-                SealedChoiceSchema sealedChoiceSchema when type.IsFrameworkType => AddAllowedValues(description, sealedChoiceSchema.Choices),
+                InputEnumType choiceSchema when type.IsFrameworkType => AddAllowedValues(description, choiceSchema.AllowedValues),
                 _ => description
             };
 
-            static FormattableString AddAllowedValues(FormattableString description, ICollection<ChoiceValue> choices)
+            static FormattableString AddAllowedValues(FormattableString description, IReadOnlyList<InputEnumTypeValue> choices)
             {
                 var allowedValues = string.Join(" | ", choices.Select(c => c.Value).Select(v => $"\"{v}\""));
```

Comment:
I think we could do better here by joining them using the `Join` method in our `FormattableStringHelper` static class because here we are assuming the allowed values here are all strings (we are adding quotes), but actually, we could do that using
```csharp
.Select(v => (FormattableString)$"{v:L}")
```
and then joining together to build another `FormattableString` instance using `FormattableStringHelper.Join` method, in this way, the write will automatically handle all values in literal, which adds quotes when it is a string, and handles correctly for other type of values.

Reply:
Created https://github.com/Azure/autorest.csharp/pull/4501 for this simplification.